src/main/msp/msp.c

```diff
@@ -2455,6 +2458,11 @@ static mspResult_e mspProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 
     case MSP_SET_RSSI_CONFIG:
         rxConfigMutable()->rssi_channel = sbufReadU8(src);
+        if (featureIsEnabled(FEATURE_RSSI_ADC) && rxConfig()->rssi_channel) {
```

Comment:
These checks should be done in `validateAndFixConfig()` in `fc/config.c`.

Reply:
Thanks for the tip.  Will update.

Reply:
Updated -- I used a bit of msp.c like before, though.  My rationale for keeping part of it around was to avoid forcing users to perform a manual power cycle in some scenarios.  I have implemented the feature check in config.c as part of `validateAndFixConfig()` but forced a `saveConfigAndNotify()` to run the validation as well as update the RSSI source.  This updates the eeprom with validation and updates the OSD widget successfully without requiring a manual power cycle.

Reply:
I suppose it may be possible to do both in feature.c .. will explore more.